### PR TITLE
Breaking: Change no-space-before-semi to semi-spacing and add "after" option (fixes #1671)

### LINKED ISSUE #1671: rule suggestion: space-after-semi
What about a rule for enforcing (or not) a space after a semicolon, as long as it's not at the end of a line? Just to separate blocks, as in 

```js
for (i = 0; i < l; i++) {
  x++;
}
foo++; bar = some_function(baz);
```

instead of 

```js
for (i = 0;i < l;i++) {
  x++;
}
foo++;bar = some_function(baz);
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/7855167-rule-suggestion-space-after-semi?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by lo1tuma**

### CODE REVIEW

File: docs/rules/semi-spacing.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+## Rule Details
+
+This rule aims to enforce spacing around a semicolon. This rule prevents the use of spaces before a semicolon in expressions.
+
+### Options
+
+The rule takes one option, an object, which has two keys `before` and `after` having boolean values `true` or `false`.
+If `before` is `true`, space is enforced before semicolons and if it's `false`, space is disallowed before semicolons.
+If `after` is `true`, space is enforced after semicolons and if it's `false`, space is disallowed after semicolons.
```
Comment (lo1tuma) [inline docs/rules/semi-spacing.md]: We should mention here that `after` is only applied when the semicolon is not the last token in its line.

**Comment by nzakas** [comment]
LGTM. That one doc change is all I have.

**Comment by lo1tuma** [comment]
I've updated the documentation.

**Comment by nzakas** [comment]
:+1:

**Pull request merged**